runtime/parachains/src/configuration.rs

```diff
@@ -1104,21 +1192,146 @@ mod tests {
+			assert_eq!(<Configuration as Store>::PendingConfigs::get(), vec![(2, config.clone())]);
 
-			Configuration::initializer_on_new_session(&2);
+			on_new_session(2);
 
 			assert_eq!(Configuration::config(), config);
-			assert_eq!(<Configuration as Store>::PendingConfig::get(3), None);
+			assert_eq!(<Configuration as Store>::PendingConfigs::get(), vec![]);
 		})
 	}
 
+	#[test]
+	fn consecutive_changes_within_one_session() {
+		new_test_ext(Default::default()).execute_with(|| {
+			let old_config = Configuration::config();
+			let mut config = old_config.clone();
+			config.validation_upgrade_delay = 100;
+			config.validation_upgrade_frequency = 100;
+			assert!(old_config != config);
+
+			assert_ok!(Configuration::set_validation_upgrade_delay(Origin::root(), 100));
+			assert_ok!(Configuration::set_validation_upgrade_frequency(Origin::root(), 100));
+			assert_eq!(Configuration::config(), old_config);
+			assert_eq!(<Configuration as Store>::PendingConfigs::get(), vec![(2, config.clone())]);
+
+			on_new_session(1);
+
+			assert_eq!(Configuration::config(), old_config);
+			assert_eq!(<Configuration as Store>::PendingConfigs::get(), vec![(2, config.clone())]);
+
+			on_new_session(2);
+
+			assert_eq!(Configuration::config(), config);
+			assert_eq!(<Configuration as Store>::PendingConfigs::get(), vec![]);
+		});
+	}
+
+	#[test]
+	fn pending_next_session_but_we_upgrade_once_more() {
+		new_test_ext(Default::default()).execute_with(|| {
+			let initial_config = Configuration::config();
+			let intermediate_config =
+				HostConfiguration { validation_upgrade_delay: 100, ..initial_config.clone() };
+			let final_config = HostConfiguration {
+				validation_upgrade_delay: 100,
+				validation_upgrade_frequency: 99,
+				..initial_config.clone()
+			};
+
+			assert_ok!(Configuration::set_validation_upgrade_delay(Origin::root(), 100));
+			assert_eq!(Configuration::config(), initial_config);
+			assert_eq!(
+				<Configuration as Store>::PendingConfigs::get(),
+				vec![(2, intermediate_config.clone())]
+			);
+
+			on_new_session(1);
+
+			// We are still waiting until the pending configuration is applied and we add another
+			// update.
+			assert_ok!(Configuration::set_validation_upgrade_frequency(Origin::root(), 99));
+
+			// This should result in yet another configiguration change scheduled.
+			assert_eq!(Configuration::config(), initial_config);
+			assert_eq!(
+				<Configuration as Store>::PendingConfigs::get(),
+				vec![(2, intermediate_config.clone()), (3, final_config.clone())]
+			);
+
+			on_new_session(2);
+
+			assert_eq!(Configuration::config(), intermediate_config);
+			assert_eq!(
+				<Configuration as Store>::PendingConfigs::get(),
+				vec![(3, final_config.clone())]
+			);
+
+			on_new_session(3);
+
+			assert_eq!(Configuration::config(), final_config);
+			assert_eq!(<Configuration as Store>::PendingConfigs::get(), vec![]);
+		});
+	}
+
+	#[test]
+	fn test() {
```

Comment:
That test name is not super descriptive.

Reply:
Oops, forgot to rename!